test/integrationtests/step_impl/test_helpers.py

```
@@ -0,0 +1,66 @@
+import time
+
+import boto3
+import botocore
+from getgauge.python import Messages
+
+from .tf_aws_resources import get_terraform_output
+from utils import get_datetime_now
+
+REGION_NAME = "eu-west-2"
+dynamodb = boto3.resource("dynamodb", REGION_NAME)
+s3 = boto3.client("s3", REGION_NAME)
+stepfunctions = boto3.client("stepfunctions", REGION_NAME)
+
+
+def await_s3_object_exists(bucket, key):
+    try:
+        waiter = s3.get_waiter("object_exists")
+        waiter.wait(Bucket=bucket, Key=key)
+        Messages.write_message(f"Object exists: {bucket}/{key}")
+        return True
+    except botocore.exceptions.WaiterError:
+        Messages.write_message(f"Object not found after waiting: {bucket}/{key}")
+        return False
```

Maybe raise an exception from this since it would be a fail for it not to succeed

---

Discussed with James to leave as is as it's consistent with the test code style